dbc_teacher.rb

```diff
@@ -0,0 +1,38 @@
+require_relative 'dbc_member'
+
+class DbcTeacher < DbcMember
+  attr_reader :salary, :target_raise
+  def initialize(options={})
+    super
+    @phase = 3
+  end
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def salary=(new_salary)
+    puts "This better be good!"
+    @salary = new_salary
+  end
+
+  def receive_raise(raise)
+    @salary += raise
+  end
+
+  def set_performance_rating(rating)
+    response = ""
+    if rating > 90 && self.class == Teacher
+      receive_raise(@target_raise)
+      response = "Yay, I'm a great employee!"
+    elsif rating > 80 && self.class == ApprenticeTeacher
```

Comment:
It's mostly a matter of whether it's ever likely to be changed on a case-by-case basis.  I.e. in this case, if all teachers will always need to get the same rating, it's fine and even preferable for it to be a constant.  If one particular teacher will sometimes need to be held to a higher standard (i.e. once you're in a certain wage bracket or seniority level you need to hit a higher target rating to get another raise or something) then you want it to be an instance variable so that you can use the value for that particular teacher.